index.bs

```diff
@@ -633,6 +651,24 @@ spec:css-syntax-3;
         a user has just clicked "sign-in" for example, then they won't be surprised or confused to
         see a [=credential chooser=] if necessary.
 
+    :   <dfn>conditional</dfn>
+    ::  Discovered credentials are presented to the user in a non-modal dialog along with an
+        indication of the origin which is requesting credentials. If the user makes a gesture
+        outside of the dialog, the dialog closes without having the {{CredentialsContainer/get()}}
+        method return to the caller and without causing a user-visible error condition. If the user
+        makes a gesture that selects a credential, that credential is returned to the caller.  The
+        [=prevent silent access flag=] is treated as being `true` regardless of its actual value:
+        the {{CredentialMediationRequirement/conditional}} behavior always involves [=user
+        mediation=] of some sort if applicable credentials are discovered.
+
+        If no credentials are discovered, the user agent MAY prompt the user to take action in a way
+        that depends on the type of credential (e.g. to insert a device containing credentials).
+        Either way, the `get()` method MUST NOT return immediately with `null` to avoid revealing
+        the lack of applicable credentials to the website.
```

Comment:
:eyes: 

I think it's fine since the whole not returning empty-handed is critical to Conditional UI anyway. Either way I changed "return immediately" by "resolve immediately".

Reply:
> Where does the "MAY prompt" happen?

I'm not sure I understand this.

> It seems like it'd fit just before the call to "ask the user to choose a Credential,

We could definitely add a note there for credential types other than `PublicKeyCredential`.

> that doesn't call back to the particular credential types in the list.

WebAuthn [loosely specifies its own credential chooser](https://w3c.github.io/webauthn/#:~:text=Prompt%20the%20user%20to%20select,by%20the%20user%20agent%20otherwise.) on [`[[PublicKeyCredential/DiscoverFromExternalSource]]`](https://pr-preview.s3.amazonaws.com/nsatragno/webauthn/pull/1576.html#sctn-discover-from-external-source), and there's a bunch of conditional mediation logic I'm adding there. The implementation matches the spec in a non obvious way: only one credential type is supported at a time, so CredMan's "ask the user to choose a Credential" does not actually prompt the user ("PublicKeyCredential" is always selected as the default) and the webauthn "picker" is shown instead (and the user being prompted to interact with their authenticator satisfies the mediation requests).

I think the way this PR deals with choosing credentials (and the spec before this PR) is correct, or at least good enough. Hopefully that makes sense?